internal/observability/call.go

```diff
@@ -223,6 +231,7 @@ func (c call) endLogs(
 	}
 
 	fields := c.fields[:0]
+	fields = c.addMoreFieldsFromTransportRequest(fields)
```

Comment:
I am not sure to understand exactly why you added a new methods `c.addMoreFieldsFromTransportRequest` for this? Why not simply having:

```
fields := c.fields[:0]
fields = append(fields, zap.String("sourceProcedure", c.req.CallerProcedure))
fields = append(fields, zap.String("rpcType", c.rpcType.String()))
```

I can see you are also using the same method in the streaming log functions, but having a function here seems a bit extreme and is also a different pattern than the current way we add those fields (like rpcType). Did I miss something?

Reply:
I wanted to keep the field extraction from transport request in a single place to avoid duplicating the same code in multiple places. I did not include "rpcType" and other fields because they don't belong to transportRequest. Do you think any performance impact this may have or just the regrouping comment?
Also, do you feel very strong about this comment?

Reply:
No there will be not major performance impact from this. Heap will be the same.

It is purely esthetic for the readability of the code, with your solution people reading this will need to jump to another function (which is 1 line of code) to understand the flow. IMO readability wins in comparison with the gain of the refactoring (1 line of code).

Reply:
I agree with @Dogild . Also to be consistent to all the methods below, we should just do it inline like @Dogild suggested.